add a PR template (#3886)
pymc-devs/pymc: .github/PULL_REQUEST_TEMPLATE.md

**Thank your for opening a PR!**

Depending on what your PR does, here are a few things you might want to address in the description:

> + [ ] what are the (breaking) changes that this PR makes?

Reviewer: do we need this? I almost never see this happen.

Author: Changes include new features... And I do think we have breaking changes, and to make matters worse, they are often quite subtle: different initializations, deprecations, changes to default values, re-naming kwargs etc.

Reviewer: fair

> + [ ] important background, or a details about the implementation

Reviewer: ```suggestion
+ [ ] background about the implementation: what, why, how?
```

+ [ ] are the changes covered by tests? 

> + [ ] are example-notebooks affected in a relevant way?

Reviewer: should also add whether doc-strings and examples are added if it's a new feature.

Author: good point - I re-formulated point 3 and 4 to cover them

Reviewer: I think the point we should add is whether a potential new feature is properly documented or not.

> + [ ] right before it's ready to merge, mention the PR in the RELEASE-NOTES.md

Reviewer: ```suggestion
+ [ ] describe the change and link to the PR in the RELEASE-NOTES.md
```
